### DISCUSSION TITLE
Should I split per redis database or queue name?

### DISCUSSION THREAD

**Discussion by Colin-b** [rq/rq#1612] [Q&A]
Hi,

From what I understand, any RQ worker will pick up any queue job in the corresponding redis database.
If I have 2 different source codes (different versions of the same app or simply different applications), what would be the best approach for RQ to avoid a worker based on source code A to pick up a task put in queue by source code B?

Should I rely on queue names or use a specific redis database per source code?

Do you advise another way?

Thanks again

**Comment by ccrvlh** [comment]
Even though this shouldn't be too common in production, where the most likely scenario is to have a dedicated Redis instance / Database for RQ and a specific codebase, I do think this is very relevant, specially during development.

I don't think this is supported as of now, since all prefixes (`rq:job`, `rq:queue` etc) are hardcoded.
The only somewhat tricky part to implement this is that this is hardcoded across each class (`Queue`, `Job`, `Worker`).

A few potential implementations I can think of are:

(1) Set up a namespace at the queue level, and make every other object ask the queue what namespace to use
(2) Have a global config to set namespaces (eg: envvars)
(3) Repeat the configuration on Queue, Job, Worker

@Colin-b have you given some though on how this could be implemented?

Related to https://github.com/rq/rq/issues/182 also.
